glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -277,9 +278,12 @@ open class GleanInternalAPI internal constructor () {
         return this.gleanDataDir
     }
 
-    fun collect(pingName: String) {
-        val s = LibGleanFFI.INSTANCE.glean_ping_collect(handle, pingName)!!
-        LibGleanFFI.INSTANCE.glean_str_free(s)
+    /**
+     * Collect a ping and return a string
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    internal fun collect(ping: PingType): String? {
+        return LibGleanFFI.INSTANCE.glean_ping_collect(handle, ping.handle)?.getAndConsumeRustString()
```

Comment:
that means we can close https://bugzilla.mozilla.org/show_bug.cgi?id=1552479

Reply:
Yeah -- I initially thought we'd take it out, but it's super useful as a test-only function.  I say we close that bug when this PR lands.